lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -605,25 +605,26 @@ def _save_ssh_host_keys(self, filename: str) -> None:
             return
 
         path = os.path.expanduser("~/.ssh")
-        makedirs_safe(path)
+        makedirs_safe(path, mode=0o700)
 
         with open(filename, 'w') as f:
 
+            keys_added_by_ansible = []
```

Comment:
Adding 'mode' makes sense, but why add these extra changes? specially since they don't seem to fix any issues.

Reply:
I raised a question above: "Is it necessary to optimize _save_ssh_host_keys() in paramiko_ssh.py or was there any concern at that time?" when i read code in _save_ssh_host_keys(), i saw two almost same for loop in it, they scan self.ssh._host_keys twice, so why not use a list to save the data that will be appended at the bottom? then we only scan self.ssh._host_keys once. This changes are only for review purposes, if you think this is unnecessary, i can make a new pr which only fix permissions. Thanks for your reply.

Reply:
> Adding 'mode' makes sense, but why add these extra changes? specially since they don't seem to fix any issues.

I raised a question above: "Is it necessary to optimize _save_ssh_host_keys() in paramiko_ssh.py or was there any concern at that time?". When i read code in _save_ssh_host_keys(), i saw two almost same for loop in it, they scan self.ssh._host_keys twice, so why not use a list to save the data that will be appended at the bottom? then we only scan self.ssh._host_keys once. This changes are only for review purposes, if you think this is unnecessary, i can make a new pr which only fix permissions. Thanks for your reply.